docs/reference/synonyms/apis/synonyms-apis.asciidoc

```diff
@@ -0,0 +1,31 @@
+[[synonym-management-apis]]
+== Synonym Management APIs
+
+beta::[]
+
+++++
+<titleabbrev>Synonyms Management APIs</titleabbrev>
+++++
+
+---
+
+Synonyms Management APIs let you create and manage synonyms sets.
```

Comment:
May be we should explain what is a synonyms set with some introduction, with something like this:

"In the past there were only two ways to manage synonyms: providing synonyms inline while defining an analyzer or providing synonyms in a file.  Both these methods have their drawbacks, the former could lead to the out-blown mappings, while the later requires keep synonym files consistent across all nodes.

Synonym Management API is a new way to define and manage synonyms through an API where synonyms will be stored in an internal system index.  A number of synonyms can be grouped in a "synonyms set", and you can have as many sets as needed"....

Reply:
I like this, but I think this should be part of a blog post explaining the past situation vs the present one. I'm not sure about including the reasons for an API to be created, and talking about "new" might get stale in a few releases from now...

Reply:
I think we can still extract the key points from @mayya-sharipova's suggestion, without "time stamping" the content. For example, we can still outline the relative merits of this new API approach, like how it avoids problems from existing approaches. :)

Reply:
I like the idea of providing a value statement here, and then saving the full in-depth description for a blog post. 👍

Reply:
Thanks for the feedback! How about this? It's a small rephrasing from Mayya's statement that provides the value proposition and the comparison to other methods. LMKWy'allT!

> Synonym Management API provides a way to define and manage synonyms through an API where synonyms are stored in an internal system index. Related synonyms can be grouped in a "synonyms set". As many synonyms sets can be created as you need for managing synonyms.
> 
> This provides an alternative to the other synonyms definition methods:
> - Inline synonyms in an analyzer definition. This can lead to out-blown mappings.
> - Synonyms file. It requires to keep synonyms files consistent across all nodes.

Reply:
WDYT about these tweaks?

> The synonyms management API provides a convenient way to define and manage synonyms in an internal system index. Related synonyms can be grouped in a "synonyms set". As many synonyms sets can be created as you need for managing synonyms.
> 
> This provides an alternative to the other synonyms definition methods:
> 
> Defining inline synonyms in an analyzer definition, which can lead to [mappings explosion](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-explosion.html) and impact performance 
> Manually uploading a synonyms file and managing consistency across all nodes


On a side note we should probably also update [the synonym token filter doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-synonym-tokenfilter.html) to reference the API as well

Reply:
Nice! I've put all this in the blender and updated the PR. Thanks for your suggestions!

> On a side note we should probably also update [the synonym token filter doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-synonym-tokenfilter.html) to reference the API as well

For sure - that's something I want to tackle separately, as I'd like to check how to reorg the different synonyms definition methods, potentially as a separate guide. I'll ask for the doc teams help on this 👍